presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveStorageFormats.java

```diff
@@ -1,8 +1,284 @@
+        String createTable = format(
+                "CREATE TABLE %s(" +
+                        "   orderkey      BIGINT," +
+                        "   partkey       BIGINT," +
+                        "   suppkey       BIGINT," +
+                        "   linenumber    BIGINT," +
+                        "   quantity      BIGINT," +
+                        "   extendedprice DOUBLE," +
+                        "   discount      DOUBLE," +
+                        "   tax           DOUBLE," +
+                        "   linestatus    VARCHAR," +
+                        "   shipinstruct  VARCHAR," +
+                        "   shipmode      VARCHAR," +
+                        "   comment       VARCHAR," +
+                        "   returnflag    VARCHAR" +
+                        ") WITH (format='%s', partitioned_by = ARRAY['returnflag'])",
+                tableName,
+                storageFormat.getName());
+        query(createTable);
+
+        // INSERT INTO TABLE
+        String insertInto = format("INSERT INTO %s " +
+                "SELECT " +
+                "orderkey, partkey, suppkey, linenumber, quantity, extendedprice, discount, tax, " +
+                "linestatus, shipinstruct, shipmode, comment, returnflag " +
+                "FROM tpch.%s.lineitem", tableName, TPCH_SCHEMA);
+        query(insertInto);
+
+        // SELECT FROM TABLE
+        assertSelect("select sum(tax), sum(discount), sum(length(returnflag)) from %s", tableName);
+
+        // DROP TABLE
+        query(format("DROP TABLE %s", tableName));
+    }
+
+    @Test(dataProvider = "storage_formats", groups = {STORAGE_FORMATS})
+    public void testCreatePartitionedTableAs(StorageFormat storageFormat)
+    {
+        setSessionProperties(storageFormat);
+
+        String tableName = "storage_formats_test_create_table_as_select_partitioned_" + storageFormat.getName().toLowerCase();
+
+        // DROP TABLE
+        query(format("DROP TABLE IF EXISTS %s", tableName));
+
+        // CREATE TABLE AS SELECT
+        String createTableAsSelect = format(
+                "CREATE TABLE %s WITH (format='%s', partitioned_by = ARRAY['returnflag']) AS " +
+                        "SELECT " +
+                        "orderkey, partkey, suppkey, linenumber, quantity, extendedprice, discount, tax, " +
+                        "linestatus, shipinstruct, shipmode, comment, returnflag " +
+                        "FROM tpch.%s.lineitem",
+                tableName,
+                storageFormat.getName(),
+                TPCH_SCHEMA);
+        query(createTableAsSelect);
+
+        // SELECT FROM TABLE
+        assertSelect("select sum(tax), sum(discount), sum(length(returnflag)) from %s", tableName);
+
+        // DROP TABLE
+        query(format("DROP TABLE %s", tableName));
+    }
+
+    private static void assertSelect(String query, String tableName)
+    {
+        QueryResult expected = query(format(query, "tpch." + TPCH_SCHEMA + ".lineitem"));
```

Comment:
Doesn't this hardcode the tests to read from the TPCH connector instead of reading from the table you just created?